packageurl.go

```diff
@@ -341,3 +345,26 @@ func typeAdjustName(purlType, name string) string {
 func validQualifierKey(key string) bool {
 	return QualifierKeyPattern.MatchString(key)
 }
+
+// Make any purl type-specific adjustments to the url encoding.
+// See https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#character-encoding
```

Comment:
Looking at the spec, I see it has these bullets:

* the '@' version separator must be encoded as %40 elsewhere
* the '?' qualifiers separator must be encoded as %3F elsewhere
* the '=' qualifiers key/value separator must NOT be encoded
* the '#' subpath separator must be encoded as %23 elsewhere

It would seem like maybe things should _not_ be path escaped, but those 3 characters (`@`, `?`, and `#`) should be? WDYT @houdini91 ?

Reply:
What should we do with the space ascii char?
Are there any other ascii chars we should encode?

Reply:
It does not say spaces need to be encoded; I don't know offhand the ramifications if it is encoded. I would just follow the rules they spell out if it says "must be encoded ... elsewhere" add to the function, otherwise just directly output. We can also add tests for these cases to make sure decoding works properly (I suspect it will) but it might be good to just check we have the expected values for the `Name`

Reply:
The reason that i added the space is because the test presented in the project including this use case (and the test expected it to be encoded)